### PR TITLE
[CYBRSOURCE-36] App review

### PR DESCRIPTION
**Pull request by Arturo777**
- Logging optimization
- GraphQL mutation security

[Workspace to test](https://cybersourcereports--sandboxusdev.myvtex.com/)

### CODE REVIEW

**Review by btalma — DISMISSED** [review]
In VtexApiService.cs line 1430 ( "createPaymentResponse.PaymentId", createPaymentResponse.PaymentId ) 
createPaymentResponse may be null.  I suggest  ( "createPaymentResponse", JsonConvert.SerializeObject(createPaymentResponse) )

**Review by btalma — APPROVED** [review]

File: dotnet/Services/CybersourcePaymentService.cs [hunk 1]
```diff
@@ -48,585 +47,593 @@ public CybersourcePaymentService(IIOServiceContext context, IVtexEnvironmentVari
...
+                            taxAmount = createPaymentRequest.MiniCart.TaxValue.ToString(),
+                            freightAmount = createPaymentRequest.MiniCart.ShippingValue.ToString()
+                        },
+                        billTo = new BillTo
+                        {
+                            firstName = createPaymentRequest.MiniCart.Buyer.FirstName,
+                            lastName = createPaymentRequest.MiniCart.Buyer.LastName,
+                            address1 = $"{createPaymentRequest.MiniCart.BillingAddress.Number} {createPaymentRequest.MiniCart.BillingAddress.Street}",
+                            address2 = createPaymentRequest.MiniCart.BillingAddress.Complement,
+                            locality = createPaymentRequest.MiniCart.BillingAddress.City,
```
Comment (aurorahey) [inline dotnet/Services/CybersourcePaymentService.cs:164]: locality = createPaymentRequest.MiniCart.BillingAddress.City ?? createPaymentRequest.MiniCart.BillingAddress.Neighborhood,

File: dotnet/Services/CybersourcePaymentService.cs [hunk 2]
```diff
@@ -48,585 +47,593 @@ public CybersourcePaymentService(IIOServiceContext context, IVtexEnvironmentVari
...
+                            phoneNumber = createPaymentRequest.MiniCart.Buyer.Phone
+                        },
+                        shipTo = new ShipTo
+                        {
+                            address1 = $"{createPaymentRequest.MiniCart.ShippingAddress.Number} {createPaymentRequest.MiniCart.ShippingAddress.Street}",
+                            address2 = createPaymentRequest.MiniCart.ShippingAddress.Complement,
+                            administrativeArea = GetAdministrativeArea(createPaymentRequest.MiniCart.ShippingAddress.State, this.GetCountryCode(createPaymentRequest.MiniCart.ShippingAddress.Country)),
+                            country = this.GetCountryCode(createPaymentRequest.MiniCart.ShippingAddress.Country),
+                            postalCode = createPaymentRequest.MiniCart.ShippingAddress.PostalCode,
+                            locality = createPaymentRequest.MiniCart.ShippingAddress.City,
```
Comment (aurorahey) [inline dotnet/Services/CybersourcePaymentService.cs:178]: locality = createPaymentRequest.MiniCart.ShippingAddress.City ?? createPaymentRequest.MiniCart.ShippingAddress.Neighborhood,

File: dotnet/Services/CybersourcePaymentService.cs [hunk 3]
```diff
@@ -785,176 +831,195 @@ public async Task<RefundPaymentResponse> RefundPayment(RefundPaymentRequest refu
...
+                            {
+                                totalAmount = sendAntifraudDataRequest.Value.ToString()
+                            },
+                            billTo = new BillTo
+                            {
+                                firstName = sendAntifraudDataRequest.MiniCart.Buyer.FirstName,
+                                lastName = sendAntifraudDataRequest.MiniCart.Buyer.LastName,
+                                address1 = $"{sendAntifraudDataRequest.MiniCart.Buyer.Address.Number} {sendAntifraudDataRequest.MiniCart.Buyer.Address.Street}",
+                                address2 = sendAntifraudDataRequest.MiniCart.Buyer.Address.Complement,
+                                locality = sendAntifraudDataRequest.MiniCart.Buyer.Address.City,
```
Comment (aurorahey) [inline dotnet/Services/CybersourcePaymentService.cs:865]: locality = sendAntifraudDataRequest.MiniCart.Buyer.Address.City ?? sendAntifraudDataRequest.MiniCart.Buyer.Address.Neighborhood,

**Review by aurorahey — APPROVED** [review]

**Pull request merged**